All complexities and failure senarios are removed, No more dependancies

jobinau opened the pull request:
This is a fix to simplify the logic while addressing many of the current issues and failure scenarios
* No more port forwarding.
* No more password authentication.
* No more dependency on `psql` outside the PostgreSQL pod.
* No more dependency on host machine and pg client versions 
Fixes https://perconadev.atlassian.net/browse/PT-2233 .

jobinau committed:
All complexities and failure senarios are removed, No more dependancy on psql outside the pod, No dependancy on password authentication or portforwaring.

it-percona-cla commented:
[![CLA assistant check](https://cla-assistant.percona.com:443/pull/badge/signed)](https://cla-assistant.percona.com:443/percona/percona-toolkit?pullRequest=764) <br/>All committers have signed the CLA.

svetasmirnova reviewed (commented):
We still need to redirect STDERR of the curl command to /dev/null.

You did not set --kubeconfig for the nested kubectl call.

jobinau committed:
Additonally accomodated the review comments on PR : https://github.com/percona/percona-toolkit/pull/764, further simplified the code to remove the extra CASE switch

jobinau commented:
Hi @svetasmirnova , Thanks for the review comments.
Both points are accommodated.
Code is simplified by merging "pg" and "pgv2" into a single CASE switch condition.
I completed a few rounds of testing and found it is working as expected.

svetasmirnova committed:
PT-2233 - pt-k8s-debug-collector needs psql in the host node

- Adjusted test TestPT_2169, because it does not depend on port for
  --resource=pg anymore
